### ISSUE TITLE
Is clusterresourcebinding.name suitable by kind and name

### ISSUE THREAD

**Issue by weilaaa** [karmada-io/karmada#3627]
I have two different resources, the Kind is the same, but Group and Version are not. Those as bellow:

```
clusters.cluster.example1.com named cluster.a
clusters.cluster.example2.com named cluster.a
```

Now, I noticed clusterresourcebinding.name is generated by kind and name which means for `clusters.cluster.example1.com` and `clusters.cluster.example2.com`, the name of their clusterresourcebinding is the same, both are cluster.a-cluster. This seems to be problematic.

**Comment by weilaaa** [comment]
@RainbowMango cc

**Comment by chaunceyjiang** [comment]
Yes, This is a known issue. refer to https://github.com/karmada-io/karmada/pull/2969

**Comment by RainbowMango** [comment]
Hi @weilaaa glad to see you are back.
Yes, it's a known issue and also a tricky one. We don't have a better solution for it yet. :(

**Comment by weilaaa** [comment]
I ask is there a way to migrate the stock data to the data named by `name-hash` without affecting the user's use. 

For example, we clone the corresponding new RB based on the existing RB, and mark the existing RB as deprecated. At this time, the newly added RB will use the new `name-hash` name format. Just in case, we can also record `CollisionCount` in the annotation to prevent hash collisions. Besides that, `GenerateBindingName` does only one thing, simply generating the name is like [K8s generating the bash suffix](https://github.com/kubernetes/kubernetes/blob/2eb4eac5817f3c1e6407c11158cf460e9d4a72bf/pkg/controller/controller_utils.go#L1160).

**Comment by RainbowMango** [comment]
> I ask is there a way to migrate the existing RB data to the new RB named by name-hash without affecting the user's use.

Yes, this is also the thing that concerns me the most. I expect a solution that could transform the legacy RB without perception.

> For example, we clone the corresponding new RB based on the existing RB, and mark the existing RB as deprecated. At this time, the newly added RB will use the new name-hash name format.

Sounds good to me. Speaking of the naming rule of RB, the naming of "Work" also has similar issues. I analyzed the issue and draft some ideas at [this document](https://docs.google.com/document/d/1GbPBpXOOTza39eFECWEXz5Pi6Wok7oE11MEp-K_3pVE/edit#), you can take a look.

This is a very challenging task, and now is also a very good time to solve this problem(Now is the beginning of the v1.7 version cycle.) I'll appreciate it if you want to lead the effort.

**Comment by weilaaa** [comment]
Thanks for the chance. I'll do some investigation first to check out the impact and possible solutions.

**Comment by RainbowMango** [comment]
I'm trying to figure out which PR/Issue should be included in the coming v1.7 release which is planned at the end of this month.
I guess we don't have enough time for this feature, so I'm moving this to v1.8.

FYI, We are working on this, the relevant topic was talked at the community meeting, and see #4000 for the progress.
